txscript/reference_test.go

```diff
@@ -362,17 +435,16 @@ func testScripts(t *testing.T, tests [][]interface{}, useSigCache bool) {
 		// Generate a transaction pair such that one spends from the
 		// other and the provided signature and public key scripts are
 		// used, then create a new engine to execute the scripts.
-		tx := createSpendingTx(scriptSig, scriptPubKey)
-		vm, err := NewEngine(scriptPubKey, tx, 0, flags, sigCache)
+		tx := createSpendingTx(witness, scriptSig, scriptPubKey,
+			int64(inputAmt))
+		vm, err := NewEngine(scriptPubKey, tx, 0, flags, sigCache, nil,
+			int64(inputAmt))
 		if err == nil {
 			err = vm.Execute()
 		}
 
 		// Ensure there were no errors when the expected result is OK.
 		if resultStr == "OK" {
-			if err != nil {
```

Comment:
That can't be removed since it would cause tests which should be valid (a.ka "OK") but actually failed to not be detected.